app/code/community/Bolt/Boltpay/Model/Payment.php

```diff
@@ -975,4 +969,63 @@ protected function isCaptureRequest($newTransactionStatus, $prevTransactionStatu
         return $newTransactionStatus == self::TRANSACTION_COMPLETED ||
               ($newTransactionStatus == self::TRANSACTION_AUTHORIZED && $prevTransactionStatus == self::TRANSACTION_AUTHORIZED);
     }
+
```

Comment:
I have some serious reservations about this series of methods.  The primary case is very simple for cancelled orders and has been robustly handled from what I see from v0.0.1 some nearly two years ago written by Akshaya.  Adding this level of complexity makes very little sense.

Essentially, we are trying to force an exceptional type of webhook,  a "partial cancel", into a very discretely defined and firmly established logic for a separate case, "just cancel the order - no money collected".  This approach is bound to add new errors.  Can we try to handle this as an exceptional case like here?  

https://github.com/BoltApp/bolt-magento1/blob/develop/app/code/community/Bolt/Boltpay/controllers/ApiController.php#L192-L227

If we handle this as a Bolt_Boltpay_InvalidTransitionException taking into account the previous order state of completed and the new state of cancelled, then this PR can become very simple.  In that case, we would just return a 200 OK and optionally add an updated transaction to report the partial cancel.  It is much better to add a `$isCancelForCompletedOrder` condition, and if we feel it looks cleaner, make this as a single helper method called `canAssumeHookIsHandled` .

At the same time we preserve tried an true logic for the normal use case leaving what has been unchanged for nearly two years as unchanged, avoiding regression errors.  

I will point out some problems that I see in this PR logic below.